campaign-launcher/client/src/components/CampaignInfo/index.tsx

```diff
@@ -22,105 +25,119 @@ const formatTime = (dateString: string): string => {
 
 type Props = {
   campaign: CampaignDetails;
+  isAlreadyJoined: boolean;
+  isJoinedLoading: boolean;
 };
 
-const CampaignInfo: FC<Props> = ({ campaign }) => {
+const CampaignInfo: FC<Props> = ({
+  campaign,
+  isAlreadyJoined,
+  isJoinedLoading,
+}) => {
   const [openChartModal, setOpenChartModal] = useState(false);
 
-  const isCompleted = campaign.status === 'completed';
+  const isMobile = useIsMobile();
+
   return (
     <Box
       display="flex"
-      alignItems="center"
+      alignItems={{ xs: 'flex-start', md: 'center' }}
+      flexDirection={{ xs: 'column', md: 'row' }}
       height={{ xs: 'auto', md: '40px' }}
-      gap={4}
-      flexWrap={{ xs: 'wrap', md: 'nowrap' }}
+      gap={{ xs: 3, md: 4 }}
+      width="100%"
     >
-      <CampaignTypeLabel campaignType={campaign.type} />
       <Box
         display="flex"
         alignItems="center"
-        justifyContent="center"
-        ml={-2}
-        px={2}
-        py="6px"
-        bgcolor={mapStatusToColor(
-          campaign.status,
-          campaign.start_date,
-          campaign.end_date
-        )}
-        borderRadius="4px"
-        textTransform="capitalize"
+        width={{ xs: '100%', md: 'auto' }}
+        gap={1}
+        order={1}
       >
-        <Typography
-          variant="subtitle2"
-          color={isCompleted ? 'secondary.contrast' : 'primary.contrast'}
-        >
-          {campaign.status.split('_').join(' ')}
-        </Typography>
+        <CampaignTypeLabel campaignType={campaign.type} />
+        <CampaignStatusLabel
+          campaignStatus={campaign.status}
+          startDate={campaign.start_date}
+          endDate={campaign.end_date}
+        />
+        {isMobile && (
+          <JoinCampaign
+            campaign={campaign}
+            isAlreadyJoined={isAlreadyJoined}
+            isJoinedLoading={isJoinedLoading}
+          />
+        )}
+      </Box>
+      <Box order={{ xs: 3, md: 2 }}>
+        <CampaignAddress
+          address={campaign.address}
+          chainId={campaign.chain_id}
+          withCopy
+        />
       </Box>
-      <CampaignAddress
-        address={campaign.address}
-        chainId={campaign.chain_id}
-        withCopy
-      />
-      <Box display="flex" alignItems="center" gap={1}>
-        {campaign?.start_date && campaign?.end_date && (
-          <>
-            <CalendarIcon />
-            <CustomTooltip
-              arrow
-              placement="top"
-              title={formatTime(campaign.start_date)}
-            >
-              <Typography
-                variant="subtitle2"
-                borderBottom="1px dashed"
-                sx={{ cursor: 'pointer' }}
+      <Box display="flex" alignItems="center" gap={3} order={{ xs: 2, md: 3 }}>
+        <Box display="flex" alignItems="center" gap={1}>
+          {campaign?.start_date && campaign?.end_date && (
```

Comment:
Seems like we always have dates now
```suggestion
          (
```